ChillChaser/Services/impl/AppUsageService.cs

```diff
@@ -1,21 +1,40 @@
 ﻿
+using System.ComponentModel.DataAnnotations;
 using ChillChaser.Models.DB;
+using ChillChaser.Models.Request;
 using Microsoft.EntityFrameworkCore;
+using Microsoft.EntityFrameworkCore.Query.SqlExpressions;
 
 namespace ChillChaser.Services.impl
 {
     public class AppUsageService(IAppService appService) : IAppUsageService
     {
-        public async Task AddAppUsage(CCDbContext ctx, string appName, DateTime from, DateTime to, string userId)
+        public async Task AddAppUsage(CCDbContext ctx, DateTime fromVar, DateTime to, AppSession[] sessions, string appName, string userId)
         {
             var app = await appService.CreateOrGetApp(ctx, appName);
-
-            app.AppUsage.Add(new AppUsage
-            {
-                From = from,
-                To = to,
-                UserId = userId
-            });
+            foreach (var elem in sessions){
```

Comment:
This seems like it might be slow, having to do a DB query for every element, maybe it's ok

Reply:
agree, lav det her til et kald på en måde

Reply:
Har optimeret det lidt men kan ikke se hvordan det kan gøres meget mere effektivt uden at tilføje mere data til databasen. Hvert element har sin egen start/end time som skal tjekkes.